### PR TITLE
front: fix wrong selection when updating train

### LINKED ISSUE #5135: front: train loses selection after updating it
### What happened?

In a scenario, when we update a train and save the changes, this train loses the selection in the timetable

### What did you expect to happen?

The train should remain selected after an update

### How can we reproduce it (as minimally and precisely as possible)?

1. Go to a scenario
2. Add at least 2 trains
3. Select the second one and update it
4. The selection goes to the other train

### What operating system, browser and environment are you using?

- **Browser**: Firefox v117
- **OS**: Ubutntu
- **Env**: Dev

### OSRD version (top right corner `Account` button > `Informations`)

dea017a1

### PR DESCRIPTION
**Pull request by SharglutDev**
and refacto getSimulationResults with rtk

close #5135

### CODE REVIEW

**Review by leovalais — APPROVED** [review]

File: editoast/openapi.yaml
Comment (clarani) [inline editoast/openapi.yaml:1]: Nice !! :tada:

File: front/src/applications/operationalStudies/components/Scenario/getSimulationResults.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+export default async function getSimulationResults(
+  timetable: TimetableWithSchedulesDetails,
+  selectedTrainId?: number
+) {
+  const { selectedProjection, allowancesSettings } = store.getState().osrdsimulation;
+  store.dispatch(updateIsUpdating(true));
+  const trainSchedulesIDs = timetable.train_schedule_summaries.map((train) => train.id);
+
+  if (trainSchedulesIDs && trainSchedulesIDs.length > 0) {
+    if (!selectedTrainId || trainSchedulesIDs.length === 1) {
```
Comment (clarani) [inline front/src/applications/operationalStudies/components/Scenario/getSimulationResults.ts]: You should update the selectedTrainId (with the id of the first trainSchedule) if:
- selectedTrainId is undefined
- selectedTrainId is not in trainSchedulesIds

```suggestion
    if (!selectedTrainId || !trainSchedulesIDs.includes(selectedTrainId)) {
```

File: front/src/applications/operationalStudies/components/Scenario/getSimulationResults.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      store.dispatch(updateSelectedTrainId(timetable.train_schedule_summaries[0].id));
```
Comment (clarani) [inline front/src/applications/operationalStudies/components/Scenario/getSimulationResults.ts]:
```suggestion
      store.dispatch(updateSelectedTrainId(trainSchedulesIDs[0]));
```

**Review by clarani — CHANGES_REQUESTED** [review]
Only left 1 comment :)

File: front/src/applications/operationalStudies/components/Scenario/getSimulationResults.ts (hunk 1 above)
Comment (SharglutDev) [reply front/src/applications/operationalStudies/components/Scenario/getSimulationResults.ts]: Done :)

File: front/src/applications/operationalStudies/components/Scenario/getSimulationResults.ts (hunk 2 above)
Comment (SharglutDev) [reply front/src/applications/operationalStudies/components/Scenario/getSimulationResults.ts]: Done :)

**Review by clarani — APPROVED** [review]
LGTM and tested on my local !

**Pull request merged**